packages/java/endpoint/src/test/java/com/vaadin/hilla/route/ClientRouteRegistryTest.java

```diff
@@ -170,6 +173,161 @@ public void when_developmentMode_then_loadLatestDevModeFileRoutesJsonIfNeeded_lo
         MatcherAssert.assertThat(allRoutes, Matchers.aMapWithSize(12));
     }
 
+    @Test
+    public void when_developmentMode_fileRoutesJson_MainLayoutNoChildren_then_hasMainLayout_is_true()
+            throws IOException {
+        mockDevelopmentMode();
+        String fileRoutesJson = """
+                [
+                    {
+                        "route": "",
+                        "children": []
+                    }
+                ]""";
+        FileUtils.write(
+                projectRoot.newFile("frontend/generated/file-routes.json"),
+                fileRoutesJson, StandardCharsets.UTF_8);
+        clientRouteRegistry.registerClientRoutes(deploymentConfiguration,
+                LocalDateTime.now());
+        Assert.assertTrue("Main layout should be present",
+                clientRouteRegistry.hasMainLayout());
+    }
+
+    @Test
+    public void when_developmentMode_fileRoutesJson_MainLayoutWithChildren_then_hasMainLayout_true()
+            throws IOException {
+        mockDevelopmentMode();
+        String fileRoutesJson = """
+                [
+                  {
+                    "route": "",
+                    "children": [
+                      {
+                        "route": "",
+                        "params": {},
+                        "title": "Index"
+                      }
+                    ]
+                  }
+                ]
+                """.stripIndent();
+        FileUtils.write(
+                projectRoot.newFile("frontend/generated/file-routes.json"),
+                fileRoutesJson, StandardCharsets.UTF_8);
+        clientRouteRegistry.registerClientRoutes(deploymentConfiguration,
+                LocalDateTime.now());
+        Assert.assertTrue("Main layout should be present",
+                clientRouteRegistry.hasMainLayout());
+    }
+
+    @Test
+    public void when_developmentMode_fileRoutesJsonWithoutMainLayout_then_hasMainLayout_false()
+            throws IOException {
+        mockDevelopmentMode();
+        String fileRoutesJson = """
+                [
+                    {
+                        "route": "some-route",
+                        "children": [
+                            {
+                                "route": "",
+                                "params": {},
+                                "title": "Index"
+                            }
+                        ]
+                    }
+                ]""".stripIndent();
```

Comment:
This does not look like the JSON generated in a real application when there's no root layout.
```suggestion
        String fileRoutesJson = """
                {
                    "route": "",
                    "params": {},
                    "children": [
                        {
                            "route": "some-route",
                            "params": {},
                            "title": "Some Layout",
                            "children": [
                                {
                                    "route": "",
                                    "params": {},
                                    "title": "Index"
                                }
                            ]
                        }
                    ]
                }""".stripIndent();
```

Reply:
The case we are testing here is actually not to have a layout as the parent of the all the other routes (which is translated to "Main Layout").

Anyway, I have an application with the following structure (no main layout):
```
views/
|_ some-route/
    |_ @index.tsx 
```
I get the following entry in the `file-routes.json`:
```json
[
  {
    "route": "some-route",
    "params": {},
    "children": [
      {
        "route": "",
        "params": {},
        "title": "Some Route"
      }
    ]
  }
]
```
Thus, apart from the absence of `"params": {},` and the `"title": "Some Route"` instead of "Index" (which are not important to the logic we are testing here), it seems to be a real case.